realm/realm-library/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -979,4 +981,110 @@ public void testSettingListClearsOldItems() {
 
         assertEquals(1, two.getObjects().size());
     }
+
+    @Test
+    public void testRemoveAllFromRealm() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        RealmList<Dog> dogs = owner.getDogs();
+        assertEquals(TEST_OBJECTS, dogs.size());
+
+        testRealm.beginTransaction();
+        dogs.removeAllFromRealm();
+        testRealm.commitTransaction();
+        assertEquals(0, dogs.size());
+        assertEquals(0, testRealm.where(Dog.class).count());
+    }
+
+    @Test
+    public void testRealmRemoveAllNotManagedList() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        RealmList<Dog> dogs = owner.getDogs();
+        assertEquals(TEST_OBJECTS, dogs.size());
+
+        RealmList<Dog> notManagedDogs = new RealmList<Dog>();
+        for (Dog dog : dogs) {
+            notManagedDogs.add(dog);
+        }
+
+        testRealm.beginTransaction();
+        notManagedDogs.removeAllFromRealm();
+        testRealm.commitTransaction();
+        assertEquals(0, dogs.size());
+        assertEquals(0, notManagedDogs.size());
+        assertEquals(0, testRealm.where(Dog.class).count());
+    }
+
+    @Test
+    public void testRealmRemoveAllOutsideTransaction() {
+        Owner owner = testRealm.where(Owner.class).findFirst();
+        RealmList<Dog> dogs = owner.getDogs();
+        try {
+            dogs.removeAllFromRealm();
+            fail("removeAllFromRealm should be called in a transaction.");
+        } catch (IllegalStateException e) {
+            assertEquals("Changing Realm data can only be done from inside a transaction.", e.getMessage());
+        }
+    }
+
+    @Test
+    public void testRemoveAllFromListStandaloneObjectShouldThrow() {
```

Comment:
"IllegalStateException will be thrown if the RealmList is not managed." This doesn't throw if every object is managed.
See testRealmRemoveAllNotManagedList
"IllegalStateException will be thrown if the RealmList contains standalone object." This is what this test do.